### PR TITLE
fix(exe): use the strict equality operator, and make `JSON.parse` execute correctly

### PR DESCRIPTION
**Pull request by dev-itsheng**
In the `fs.readFileSync` function, if you do not pass the second parameter, it will return a Buffer object, which will be an error when processed by the `JSON.parse` function, so I fixed it.

Also, I noticed a lot of `==` and `!=` in this file, which I think should be replaced with better `===` and `!==`.

### CODE REVIEW

**Review by zkochan — APPROVED** [review]

File: packages/exe/setup.js [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
   : process.platform
-const arch = platform == 'win' && process.arch == 'ia32' ? 'x86' : process.arch
+const arch = platform === 'win' && process.arch === 'ia32' ? 'x86' : process.arch
 
 const pkgName = `@pnpm/${platform}-${arch}`
 const pkgJson = require.resolve(`${pkgName}/package.json`)
 const subpkg = JSON.parse(fs.readFileSync(pkgJson, 'utf8'))
 
-if (subpkg.bin != null) {
+if (subpkg.bin !== null) {
```
Comment (zkochan) [inline packages/exe/setup.js]: no, this is not good. `!=` should be used because it will work with both undefined and null

Comment (dev-itsheng) [reply packages/exe/setup.js]: Yes, when I saw this, I was thinking: "There is only `null` in JSON, there will be no `undefined`", but I didn't realize until just now that if there is no corresponding property, the parsed value is also `undefined`.

Comment (dev-itsheng) [reply packages/exe/setup.js]: So, here, can we just check if the value is truthy? like this:

```javascript
if (subpkg.bin) {
  // ...
}
```

Comment (zkochan) [reply packages/exe/setup.js]: I don't think the linter will allow this. And I also don't see value in this change.

Comment (dev-itsheng) [reply packages/exe/setup.js]: Yes, I withdraw this modification.

**Comment by fisker** [comment]
> In the fs.readFileSync function, if you do not pass the second parameter, it will return a Buffer object, which will be an error when processed by the JSON.parse function

Not true https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-json-parse-buffer.md

**Comment by dev-itsheng** [comment]
> > In the fs.readFileSync function, if you do not pass the second parameter, it will return a Buffer object, which will be an error when processed by the JSON.parse function
> 
> Not true https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-json-parse-buffer.md

Yes, I researched it, it is also possible not to pass 'utf8' as the second parameter.

But there is still a little problem with the statement in the link you gave:

> When reading and parsing a JSON file, it's unnecessary to read it as a string, because JSON.parse() can also parse Buffer.
>
> Passing in a buffer may not **be performant** and is not compatible with TypeScript.

According to the [ECMA-262 standard](https://262.ecma-international.org/5.1/#sec-15.12.2), when `JSON.parse` is executed, it must first make a ` toString` handler, if a `buffer` is passed in, it will call `Buffer.prototype.toString`. Observing [Node.js related source code](https://github.com/nodejs/node/blob/master/lib/buffer.js#L784), you can find that it does use utf8 for conversion internally.

So, passing a Buffer in `JSON.parse` does not have much impact on performance, and [TypeScript Type Definition](https://github.com/microsoft/TypeScript/blob/main/lib/lib.es5.d.ts#L1059) The related problem is because when I debugged the code, I noticed that TypeScript reported an error `TS2345: Argument of type 'Buffer' is not assignable to parameter of type 'string'`. I tried to make changes.

So, next I'm going to try to modify TypeScript's type definitions and the instructions for eslint-plugin-unicorn.

thanks for reminding.

**Pull request merged**